internal/controllers/synthesis/lifecycle.go

```diff
@@ -160,34 +160,40 @@ func (c *podLifecycleController) Reconcile(ctx context.Context, req ctrl.Request
 	return ctrl.Result{}, nil
 }
 
-func (c *podLifecycleController) shouldDeletePod(logger logr.Logger, comp *apiv1.Composition, syn *apiv1.Synthesizer, pods *corev1.PodList) (logr.Logger, *corev1.Pod /* exists */, bool) {
+func shouldDeletePod(logger logr.Logger, comp *apiv1.Composition, syn *apiv1.Synthesizer, pods *corev1.PodList) (logr.Logger, *corev1.Pod, bool /* exists */) {
 	if len(pods.Items) == 0 {
 		return logger, nil, false
 	}
 
 	// Only create pods when the previous one is deleting or non-existant
+	var onePodDeleting bool
 	for _, pod := range pods.Items {
```

Comment:
Should we check the length of this list? Or document how the loop is supposed to behave? For what I can read, if the loop encounters 2 pods that are being deleted the creation of a new one is allowed, but what if the deletion takes a long time? I don't see how we're bounding the total Pod count.

Reply:
There's a comment a bit lower that attempts to provide some context. Essentially we already reduce the concurrency of _non-deleted_ pods to 1, so adding this check effectively limits worst-case concurrency to 2.